helix-core/src/movement.rs

```diff
@@ -231,6 +249,7 @@ where
     // need to +1 so that prev() includes current char
     let mut chars = slice.chars_at(*pos + 1);
 
+    #[allow(clippy::while_let_on_iterator)]
```

Comment:
Is this a false positive, maybe report it to clippy?

Reply:
Not exactly, but the next line operates on next(), and I didn't know how to get with that so I decided to surprise the warning for now.